### PR TITLE
doc(propagators): switch to use new Extensions.Propagators package, add some documentation

### PR DESCRIPTION
**Pull request by dmpe**
Continuing https://github.com/open-telemetry/opentelemetry-dotnet/pull/3244

## Changes

- [x] Add Readme.md files
- [x] Modify all existing code in this repo to switch to new propagator.
- [x] Design discussion issue https://github.com/open-telemetry/opentelemetry-dotnet/issues/1881
- [x] Changes in public API reviewed
- [ ] Obsolete the current B3Propagator  -> a new PR

### CODE REVIEW

File: src/OpenTelemetry.Extensions.Propagators/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Trace context propagators for OpenTelemetry .NET
+
+[![NuGet](https://img.shields.io/nuget/v/OpenTelemetry.Extensions.Propagators.svg)](https://www.nuget.org/packages/OpenTelemetry.Extensions.Propagators)
+[![NuGet](https://img.shields.io/nuget/dt/OpenTelemetry.Extensions.Propagators.svg)](https://www.nuget.org/packages/OpenTelemetry.Extensions.Propagators)
+
+The package provides context propagators (currently supporting B3 format) for tracing.
```
Comment (reyang) [inline src/OpenTelemetry.Extensions.Propagators/README.md:6]:
```suggestion
The package provides context propagators (currently supporting [B3](https://github.com/openzipkin/b3-propagation) format) for tracing.
```

File: src/OpenTelemetry.Extensions.Propagators/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        new OpenTelemetry.Extensions.Propagators.B3Propagator(),
+        new BaggagePropagator(),
+    }));
+```
+
+## References
+
+* [B3 (Zipkin) Context specification](https://github.com/openzipkin/b3-propagation)
+
```
Comment (reyang) [inline src/OpenTelemetry.Extensions.Propagators/README.md:46]: Minor: probably can remove this empty line

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/OpenTelemetry.Instrumentation.Grpc.Tests.csproj [hunk 3]
```diff
@@ -34,6 +34,7 @@
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Exporter.InMemory\OpenTelemetry.Exporter.InMemory.csproj" />
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Instrumentation.AspNetCore\OpenTelemetry.Instrumentation.AspNetCore.csproj" />
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Instrumentation.GrpcNetClient\OpenTelemetry.Instrumentation.GrpcNetClient.csproj" />
+    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Extensions.Propagators\OpenTelemetry.Extensions.Propagators.csproj" />
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.Grpc.Tests/OpenTelemetry.Instrumentation.Grpc.Tests.csproj:37]: nit, move this to Ln35 (alphabetic ordering)

**Review by reyang — APPROVED** [review]

**Comment by dmpe** [comment]
Hi, thanks for the feedback. Adjusted and pushed. Tests are green :)

File: src/OpenTelemetry.Extensions.Propagators/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Trace context propagators for OpenTelemetry .NET
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions.Propagators/README.md:1]: Title is not accurate...

File: src/OpenTelemetry.Extensions.Propagators/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+[![NuGet](https://img.shields.io/nuget/v/OpenTelemetry.Extensions.Propagators.svg)](https://www.nuget.org/packages/OpenTelemetry.Extensions.Propagators)
+[![NuGet](https://img.shields.io/nuget/dt/OpenTelemetry.Extensions.Propagators.svg)](https://www.nuget.org/packages/OpenTelemetry.Extensions.Propagators)
+
+The package provides context propagators (currently
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions.Propagators/README.md:6]: please add a link to spec : https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md

**Review by cijothomas — APPROVED** [review]
LGTM. small nits left, okay for follow ups!

File: src/OpenTelemetry.Extensions.Propagators/README.md (hunk 4 above)
Comment (dmpe) [reply src/OpenTelemetry.Extensions.Propagators/README.md]: I have now called that `Propagator formats for OpenTelemetry .NET` - as it is likely to include other formats in the future such as Jaeger. Does it reflect the purpose better?

**Review by cijothomas — APPROVED** [review]

**Pull request merged**